core/src/main/java/io/micronaut/core/annotation/AnnotationMetadata.java

```diff
@@ -611,6 +616,97 @@ default boolean isDeclaredAnnotationPresent(@Nonnull Class<? extends Annotation>
         return result.map(OptionalInt::of).orElseGet(OptionalInt::empty);
     }
 
+    /**
+     * The value as an {@link OptionalInt} for the given annotation and member.
+     *
+     * @param annotation The annotation
+     * @param member     The member
+     * @return THe {@link OptionalInt} value
+     */
+    default @Nonnull OptionalInt intValue(@Nonnull Class<? extends Annotation> annotation, @Nonnull String member) {
```

Comment:
I think the `@Nonnull` annotation may be redundant here. Kotlin treats Optional as non null without the annotation. I haven't tested OptionalInt, though I assume it works the same

Reply:
Thanks will remove them